src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,39 +1,70 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+
+type Props = {
+  selectedTodo: Todo;
+  setSelectedTodo: (todo: Todo | null) => void;
+};
+
+export const TodoModal: React.FC<Props> = ({
+  selectedTodo,
+  setSelectedTodo,
+}) => {
+  const [loader, setLoader] = useState(true);
+  const [user, setUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    getUser(selectedTodo.userId).then(fetchedUser => {
+      setUser(fetchedUser);
+      setLoader(false);
+    });
+  });
```

Comment:
You're using `useEffect` without a dependency array, which means it will run after every render. This could lead to performance issues and unexpected behavior since `getUser` will be called too frequently. Consider adding a dependency array with `selectedTodo.userId` to ensure it only runs when the `userId` changes.